pensieve/analysis.py

```diff
@@ -0,0 +1,95 @@
+from datetime import datetime, timedelta
+import mozanalysis
+from mozanalysis.bq import BigQueryContext
+from mozanalysis.bq import sanitize_table_name_for_bq
+from mozanalysis.experiment import TimeLimits
+import mozanalysis.metrics.desktop as mmd
+import pytz
+
+
+class Analysis:
+    """
+    Wrapper for analysing experiments.
+    """
+
+    # todo: where should this come from? experiments, metrics?
+    ANALYSIS_PERIOD = 7  # 1 week
+
+    # list of standard metrics to be computed
+    STANDARD_METRICS = [
+        mmd.active_hours,
+        mmd.uri_count,
+        mmd.ad_clicks,
+        mmd.search_count,
+    ]
+
+    def __init__(self, project, dataset):
+        self.project = project
+        self.dataset = dataset
+        self.bq_context = None
+
+    def _current_date(self):
+        """Returns the current date with UTC timezone and time set to 00:00:00."""
+        return datetime.combine(datetime.today(), datetime.min.time()).replace(tzinfo=pytz.utc)
+
+    def _should_analyse_experiment(self, experiment, current_date):
+        """
+        Returns True if a passed experiment should be analysed based
+        on its start date and the last time it got analysed.
+        """
+        date_delta = current_date - experiment.start_date
+        next_analysis_date = current_date + timedelta(days=date_delta.days % self.ANALYSIS_PERIOD)
+
+        return (
+            date_delta.days > 0 and current_date == next_analysis_date
+        ) or experiment.end_date <= current_date
+
+    def run(self, experiment):
+        """
+        Run analysis using mozanalysis for a specific experiment.
+        """
+
+        if experiment.normandy_slug is None:
+            return  # some experiments do not have a normandy slug
+
+        current_date = self._current_date()
+
+        if self._should_analyse_experiment(experiment, current_date):
+            exp = mozanalysis.experiment.Experiment(
+                experiment_slug=experiment.normandy_slug,
+                start_date=experiment.start_date.strftime("%Y-%m-%d"),
+            )
+
+            date_delta = current_date - experiment.start_date
+
+            # data from the current day aren't available yet, so we use the previous day
+            last_date_full_data = current_date - timedelta(days=1)
```

Comment:
If we use the date passed in to the container, "current_date" should be correct date and we'd have this job run after the job that populates the data this depends on

Reply:
Alright, I added `date` as a parameter that can be passed from airflow